sql-parser/src/test/java/io/crate/sql/LiteralsTest.java

```diff
@@ -47,4 +47,87 @@ public void testQuote() throws Exception {
+        assertThat(Literals.replaceEscapedChars("hello\\s"), is("hello\\s"));
+    }
+
+    @Test
+    public void testReplaceEscapedCharsOctalByteValues() throws Exception {
+        assertThat(Literals.replaceEscapedChars("\\141"), is("a"));
+        assertThat(Literals.replaceEscapedChars("\\141\\141\\141"), is("aaa"));
+        assertThat(Literals.replaceEscapedChars("\\1411"), is("a1"));
+        assertThat(Literals.replaceEscapedChars("\\0"), is("\u0000"));
+        assertThat(Literals.replaceEscapedChars("\\07"), is("\u0007"));
+        assertThat(Literals.replaceEscapedChars("\\007"), is("\u0007"));
+    }
+
+    @Test
+    public void testReplaceEscapedCharsHexByteValues() throws Exception {
+        assertThat(Literals.replaceEscapedChars("\\x61"), is("a"));
+        assertThat(Literals.replaceEscapedChars("\\x610000"), is("a0000"));
+        assertThat(Literals.replaceEscapedChars("\\xDg0000"), is("\rg0000"));
+        assertThat(Literals.replaceEscapedChars("\\x61 \\x61"), is("a a"));
+        assertThat(Literals.replaceEscapedChars("What \\x61 wonderful world"), is("What a wonderful world"));
+        assertThat(Literals.replaceEscapedChars("\\x\\x"), is("xx"));
+    }
+
+    @Test
+    public void testReplaceEscapedCharsUnicodeChars() throws Exception {
+        // 16-bit unicode
+        assertThat(Literals.replaceEscapedChars("\\u0061"), is("a"));
+        assertThat(Literals.replaceEscapedChars("\\u0061\\u0061"), is("aa"));
+        assertThat(Literals.replaceEscapedChars("\\u00610000"), is("a0000"));
+        assertThat(Literals.replaceEscapedChars("\\u0061 \\u0061"), is("a a"));
+        assertThat(Literals.replaceEscapedChars("What \\u0061 wonderful world"), is("What a wonderful world"));
+        // 32-bit unicode
+        assertThat(Literals.replaceEscapedChars("\\U00000061"), is("a"));
+        assertThat(Literals.replaceEscapedChars("\\U00000061 \\U00000061"), is("a a"));
+        assertThat(Literals.replaceEscapedChars("\\U000000610000"), is("a0000"));
+        assertThat(Literals.replaceEscapedChars("What \\U00000061 wonderful world"), is("What a wonderful world"));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testInvalidLengthEscapedUnicode16ThrowsException() throws Exception {
+        Literals.replaceEscapedChars("\\u006");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testInvalidHexEscapedUnicode16ThrowsException() throws Exception {
+        Literals.replaceEscapedChars("\\u006G");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testInvalidLengthEscapedUnicode32ThrowsException() throws Exception {
+        Literals.replaceEscapedChars("\\U0061");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testInvalidHexEscapedUnicode32ThrowsException() throws Exception {
+        Literals.replaceEscapedChars("\\U0000006G");
```

Comment:
Can we also have a `expectedException.expectMessage` here in these tests, to control the error message?